pkgs/development/web/deno/default.nix

```
@@ -23,9 +24,18 @@ rustPlatform.buildRustPackage rec {
     owner = "denoland";
     repo = pname;
     rev = "v${version}";
-    sha256 = "sha256-0S5BSXWnv4DMcc8cijRQx6NyDReg5aJJT65TeNFlkkw=";
+    hash = "sha256-0S5BSXWnv4DMcc8cijRQx6NyDReg5aJJT65TeNFlkkw=";
   };
-  cargoSha256 = "sha256-DZICb85B9pWT8bV06FYjS604RdomB5nqtR55R00CT8c=";
+  cargoHash = "sha256-7Xfnc91yQiAwAF5fvtiwnELUDb7LJeye3GtXNzYkUo8=";
+
+  cargoPatches = [
+    # resolved in 1.31.2
+    (fetchpatch {
+      name = "CVE-2023-28446.patch";
+      url = "https://github.com/denoland/deno/pull/18108/commits/d2442d27bf5bac5720bdf39341925a8592d5dd71.patch";
```

please use the merged commit
```suggestion
      url = "https://github.com/denoland/deno/commit/78d430103a8f6931154ddbbe19d36f3b8630286d.patch";
```

---

if it's an already merged PR how can you change the patch before merging?
It makes sense to pin the commit **before** the PR is merged

---

I know some maintainers who fetch the PR ref, update the patch locally, merge it locally, and push the merged commit to `origin/main`; the GitHub UI shows the PR as merged but the contents from the original PR could vary.
Sorry for the noise, I know is a very specific workflow.